docker-entrypoint.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+export CONFIG_DIR="${CONFIG_DIR:-/opt/config}"
+export CONFIG=$CONFIG_DIR/config.json
+export FORCE_BUILD_BEFORE_START="${FORCE_BUILD_BEFORE_START:-false}"
+
+# Supported verbs: install (web install), setup (interactive CLI session). Default: web install
+# TODO: constraint it using a hash set (or hash table)
+export NODEBB_INIT_VERB="${NODEBB_INIT_VERB:-install}"
+# Setup variable for backward compatibility, default: <empty>
+export SETUP="${SETUP:-}"
+
+mkdir -p $CONFIG_DIR
+chmod -fR 760 $CONFIG_DIR
+
+if [[ ! -w $CONFIG_DIR/package.json ]]; then
+  echo "panic: no write permission for $CONFIG_DIR"
+  exit 1
+fi
+
+[[ -f $CONFIG_DIR/package.json ]] || cp install/package.json $CONFIG_DIR/package.json
+[[ -f $CONFIG_DIR/package-lock.json ]] || touch $CONFIG_DIR/package-lock.json
+
+ln -fs $CONFIG_DIR/package.json package.json
+ln -fs $CONFIG_DIR/package-lock.json package-lock.json
+
+npm install --only=prod
```

Comment:
npm moved to `--omit` instead of `--only` for some reason, so this throws a deprecation warning. So this should now be
```bash
npm install --omit=dev
```
(note that this is from npm >=8, which became the default in Node 16, so it's currently in all supported node versions)

Reply:
Correction: also add `--no-optional` because `sass-embedded` currently doesn't support alpine Linux (it might soon-ish - it appears dart finally added support for musl builds a few months ago)